cmd/immuadmin/command/user.go

```diff
@@ -18,189 +18,212 @@ package immuadmin
-				row[1] = "admin"
-				row[2] = "admin"
-			} else {
-				row[1] = "client"
-				switch permission {
+	switch command {
+	case "help":
+		fmt.Println("user list  -- shows all users and their details")
+		fmt.Println()
+		fmt.Println("user create user_name permission database_name  -- creates a user for the database, asks twice for the password")
+		fmt.Println()
+		fmt.Println("user changepassword username  -- asks to insert the new password twice")
+		fmt.Println()
+		fmt.Println("user permission grant/revoke username permission_type database_name  -- grants or revokes the permission for the database")
+		fmt.Println()
+		fmt.Println("user activate/deactivate username  -- activates or deactivates a user")
+		fmt.Println()
+		return "", nil
+	case "list":
+		userlist, err := cl.immuClient.ListUsers(context.Background())
+		if err != nil {
+			return "", err
+		}
+		fmt.Println()
+		fmt.Println("User\tActive\tCreated By\tCreated At\t\t\t\t\tDatabase\tPermission")
+		for _, val := range userlist.Users {
+			fmt.Printf("%s\t%v\t%s\t\t%s\n", string(val.User), val.Active, val.Createdby, val.Createdat)
+			for _, val := range val.Permissions {
+				fmt.Printf("\t\t\t\t\t\t\t\t\t\t%s\t\t", val.Database)
+				switch val.Permission {
+				case auth.PermissionAdmin:
+					fmt.Printf("Admin\n")
+				case auth.PermissionSysAdmin:
+					fmt.Printf("System Admin\n")
 				case auth.PermissionR:
-					row[2] = "read"
-				case auth.PermissionRW:
-					row[2] = "readwrite"
-				case auth.PermissionNone:
-					row[2] = "deactivated"
+					fmt.Printf("Read\n")
+				case auth.PermissionW:
+					fmt.Printf("Write\n")
 				default:
-					row[2] = fmt.Sprintf("unknown: %d", permission)
+					return "Permission value not recognized. Allowed permissions are read,write,admin", nil
 				}
 			}
-			return row
-		},
-	)
-}
-
-func (cl *commandline) createUser(username string, permissions string) {
-	fmt.Println("NOTE:", auth.PasswordRequirementsMsg+".")
-	pass, err := cl.passwordReader.Read(fmt.Sprintf("Choose a password for %s:", username))
-	if err != nil {
-		c.QuitToStdErr(err)
-	}
-	if err = auth.IsStrongPassword(string(pass)); err != nil {
-		c.QuitToStdErr(errors.New("Password does not meet the requirements"))
-	}
-	pass2, err := cl.passwordReader.Read("Confirm password:")
-	if err != nil {
-		c.QuitToStdErr(err)
-	}
-	if !bytes.Equal(pass, pass2) {
-		c.QuitToStdErr(errors.New("Passwords don't match"))
-	}
-	var permission []byte
-	switch permissions {
-	case "readwrite":
-		permission = []byte{auth.PermissionRW}
-	default:
-		permission = []byte{auth.PermissionR}
-	}
-	_, err = cl.immuClient.CreateUser(cl.context, []byte(username), pass, permission)
-	if err != nil {
-		c.QuitWithUserError(err)
-	}
-	fmt.Printf("User %s created\n", username)
-}
+			fmt.Println()
+		}
+		return "", nil
+	case "create":
+		if len(args) < 4 {
+			return "Incorrect number of parameters for the command. Please type 'user help' for more information.", nil
```

Comment:
Could you please use:
`Incorrect number of parameters for this command. Please type...
`